packages/firestore/test/unit/specs/limbo_spec.test.ts

```diff
@@ -434,4 +434,59 @@ describeSpec('Limbo Documents:', [], () => {
         .expectEvents(query, { removed: [docC] });
     }
   );
+
+  specTest('Limbo documents stay consistent between views', [], () => {
+    // This tests verifies that a document is consistent between views, even
+    // if the document is only in Limbo in one of the.
+    const originalQuery = Query.atPath(path('collection')).addFilter(
+      filter('matches', '==', true)
+    );
+    const filteredQuery = Query.atPath(path('collection'))
+      .addFilter(filter('matches', '==', true))
+      .addFilter(filter('key', '>=', 'a'));
+
+    const docA = doc('collection/a', 1000, { key: 'a', matches: true });
+    const docACommitted = doc(
+      'collection/a',
+      1000,
+      { key: 'a', matches: true },
+      { hasCommittedMutations: true }
+    );
+    const docBCommitted = doc(
+      'collection/b',
+      1001,
+      { key: 'b', matches: true },
+      { hasCommittedMutations: true }
+    );
+
+    return (
+      client(0)
+        .userSets('collection/a', { key: 'a', matches: true })
+        .userSets('collection/b', { key: 'b', matches: true })
+        .writeAcks('collection/a', 1000)
+        .writeAcks('collection/b', 1001)
+        .userListens(originalQuery)
+        .expectEvents(originalQuery, {
+          added: [docACommitted, docBCommitted],
+          fromCache: true
+        })
+        .watchAcksFull(originalQuery, 2000, docACommitted)
```

Comment:
I don't understand why watch would only ack `docACommitted` here. Does this presuppose that docB has been modified out of band here? If so please comment.

Also a point of clarity: isn't the thing that watch is acknowledging docA?

Reply:
This makes more sense if docACommitted is just docA here (since Watch doesn't consider it dirty). I also added a comment about the out of band modification of docB, which cases it to go into Limbo.